tests/integration/collective/CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+#=============================================================================
+# Copyright 2023 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+# We abuse find package for testing purposes here to
+# 'find' the current build tree to test package builds
+
+
+cmake_minimum_required(VERSION 3.22.1 FATAL_ERROR)
+
+project(collective VERSION 1.0 LANGUAGES C CXX)
+
+set(CMAKE_CXX_STANDARD 17)
+set(BUILD_SHARED_LIBS ON)
+
+find_package(legate_core REQUIRED)
+
+legate_add_cpp_subdirectory(src TARGET collective  EXPORT collective-export)
+
+legate_add_cffi(${CMAKE_CURRENT_SOURCE_DIR}/src/collective_cffi.h TARGET collective)
+legate_python_library_template(collective)
```

Comment:
note that the python library template is going to be deprecated, and it'd be easier to directly spell out boilerplates, which isn't that much at this point. I suggest you look at other recent integration tests that do that (e.g., https://github.com/nv-legate/legate.core/tree/branch-23.07/tests/integration/tree_reduce)